tools/OpenConsole.psm1

```diff
@@ -7,29 +7,51 @@ Set-Item -force -path "env:OpenConsoleRoot" -value "$PSScriptRoot\.."
 # them into the Powershell environment.
 function Set-MsbuildDevEnvironment()
 {
-    $vswhere = ${env:ProgramFiles(x86)} + '\Microsoft Visual Studio\Installer\vswhere.exe'
-    if (-not (Test-Path $vswhere -PathType Leaf))
+    $ErrorActionPreference = 'Stop'
+
+    # VSSetup acts similarly to vswhere, but with a powershell interface
+    $need_vssetup = $null -eq (Get-Module -Name 'VSSetup')
+
+    if ($need_vssetup)
     {
-        throw "Visual Studio is not installed on this system"
+        $vssetup_path = `
+            Join-Path $env:Temp ([System.IO.Path]::GetRandomFileName())

```

Comment:
Each time they start the script you leave a bunch of files behind. Why not just use a .gitignore'd directory under src that is stable, and prepend it to $env:PSModulePath so that PowerShell does all the appropriate work, and it works without having to re-download the module each time the script starts unless the person git cleans their repo?

Reply:
Because I'm not sure of the best way to do this and I expected one of you two to tell me what I should do ;)

I'll make the changes later today.